router/internal/metric/prometheus.go

```diff
@@ -0,0 +1,57 @@
+package metric
+
+import (
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/wundergraph/cosmo/router/internal/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+	"net/http"
+	"strings"
+	"time"
+	"unicode"
+)
+
+// Excluded by default from Prometheus export because of high cardinality
+// This would produce a metric series for each unique operation
+// Metric must be in snake case because that's how the Prometheus exporter converts them
+var defaultExcludedOtelKeys = []attribute.Key{
+	otel.WgOperationHash,
+}
+
+func ServePrometheus(logger *zap.Logger, listenAddr string, path string, registry *prometheus.Registry) *http.Server {
+	r := chi.NewRouter()
+	r.Use(middleware.Recoverer)
+	r.Handle(path, promhttp.HandlerFor(registry, promhttp.HandlerOpts{
+		EnableOpenMetrics: true,
+		ErrorLog:          zap.NewStdLog(logger),
+		Registry:          registry,
+		Timeout:           0,
+	}))
+
+	svr := &http.Server{
+		Addr:              listenAddr,
+		ReadTimeout:       1 * time.Minute,
+		WriteTimeout:      2 * time.Minute,
+		ReadHeaderTimeout: 20 * time.Second,
+		ErrorLog:          zap.NewStdLog(logger),
```

Comment:
Are you sure about these timeouts?
I have no experience with Prom but these seem to be quite high.

Reply:
I just set some because by default there are none. I made them smaller